src/index.ts

```diff
@@ -60,3 +61,28 @@ export async function resolveLockfile (id: string = process.cwd(), opts: Resolve
   }
   throw new Error('No lockfile found from ' + id)
 }
+
+export async function findWorkspaceDir (id: string = process.cwd(), opts: ResolveOptions = {}): Promise<string> {
+  const resolvedPath = isAbsolute(id) ? id : await resolvePath(id, opts)
+  const _opts = { startingFrom: resolvedPath, reverse: true, ...opts }
+
+  // Lookdown for .git/config
+  try {
+    const r = await findFile('.git/config', _opts)
+    return resolve(r, '../..')
+  } catch { }
```

Comment:
I'm not sure finding farthest `.git` is advisable. Users may have a `.git` folder in their home directory for dotfiles, for example.